mart/attack/threat_model.py

```diff
@@ -5,21 +5,60 @@
 #
 
 import abc
-from typing import Any, Dict, Union
+from typing import Any, Dict, Optional, Union
```

Comment:
I think we should standardize on `from __future__ import annotations` and the use of `| None` for `Optional`.